lib/renderer/extensions/storage.js

```diff
@@ -1,88 +1,126 @@
-const getStorage = (storageType) => {
-  const data = window.localStorage.getItem(`__chrome.storage.${storageType}__`)
-  if (data != null) {
-    return JSON.parse(data)
-  } else {
-    return {}
-  }
+const fs = require('fs')
+const path = require('path')
+const { remote } = require('electron')
+const { app } = remote
+
+const getChromeStoragePath = (storageType, extensionId) => {
+  return path.join(
+    app.getPath('userData'), `/Chrome Storage/${extensionId}-${storageType}.json`)
 }
 
-const setStorage = (storageType, storage) => {
-  const json = JSON.stringify(storage)
-  window.localStorage.setItem(`__chrome.storage.${storageType}__`, json)
+// recursively create directory and parent directories if needed
+const mkdirParent = (dirPath, mode, callback) => {
+  fs.mkdir(dirPath, mode, error => {
+    if (error && error.errno === 34) {
+      fs.mkdirParent(path.dirname(dirPath), mode, callback)
+      fs.mkdirParent(dirPath, mode, callback)
+    }
+    callback && callback(error)
```

Comment:
On the worst case `callback` might be called 3 times for one `mkdirParent` call, for async API you have to wait until the last one finishes before continue.

```js

const mkdirp = (dir, callback) => {
  fs.mkdir(dir, (error) => {
    if (error && error.code === 'ENOENT') {
      mkdirp(path.dirname(dir), (error) => {
        if (!error) {
          mkdirp(dir, callback)
        }
      })
    } else if (error && error.code === 'EEXIST') {
      callback(null)
    } else {
      callback(error)
    }
  })
}
```